text/0000-javascript-module-api.md

```diff
@@ -0,0 +1,1058 @@
+depend on. This gives us atomic updates and makes sure you never have one piece
+of the framework interacting with a different piece that is inadvertently three
+versions old.
+
+What this means is that, rather than shipping `@ember/object` on npm, we'd ship
+a single `ember-source` (or something) package that includes the entire
+framework. At build time, the Ember build process would virtually map the
+`@ember/object` package to the right file inside `ember-source`. In essence, all
+of the benefits of smaller bundles without the boiling hellbroth of managing
+dependencies.
+
+That said, because this RFC is designed with an eye towards eventually
+publishing each package to npm individually, we will have that option available
+to us in the future once we determine that we can do so without causing lots of
+pain.
+
+# Addenda
+
+_(Ed. note: These tables are automatically generated from the scripts in the [codemod][codemod] repository.)_
+
+[codemod]: https://github.com/tomdale/ember-modules-codemod
+
+## Addendum 1 - Table of Module Names and Exports by Global
+
+| Before                                | After                                                                      |
+| ---                                   | ---                                                                        |
+| `Ember.$`                             | `import $ from "jquery"`                                                   |
+| `Ember.A`                             | `import { A } from "@ember/array"`                                         |
+| `Ember.Application`                   | `import Application from "@ember/application"`                             |
+| `Ember.Array`                         | `import EmberArray from "@ember/array"`                                    |
+| `Ember.ArrayProxy`                    | `import ArrayProxy from "@ember/array/proxy"`                              |
+| `Ember.Checkbox`                      | `import Checkbox from "@ember/component/checkbox"`                         |
+| `Ember.Component`                     | `import Component from "@ember/component"`                                 |
+| `Ember.ContainerDebugAdapter`         | `import ContainerDebugAdapter from "@ember/debug/container-debug-adapter"` |
+| `Ember.Controller`                    | `import Controller from "@ember/controller"`                               |
+| `Ember.DataAdapter`                   | `import DataAdapter from "@ember/debug/data-adapter"`                      |
+| `Ember.DefaultResolver`               | `import GlobalsResolver from "@ember/application/globals-resolver"`        |
+| `Ember.Enumerable`                    | `import Enumerable from "@ember/enumerable"`                               |
+| `Ember.Evented`                       | `import Evented from "@ember/object/evented"`                              |
+| `Ember.HashLocation`                  | `import HashLocation from "@ember/routing/hash-location"`                  |
```

Comment:
We're already following that pattern in this proposal with `Ember.Component` ... but not with `Ember.Object`. 😜

I think this will always come down to taste instead of hard and fast rules which is why I labeled it as a bikeshed.